### ISSUE TITLE
Prefer new notification syntax

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#151]
### What category of cop is this?:

Please try to select as few as possible.

- [ ] Deprecation - A cop that alerts to a deprecation in the Chef Infra Client
- [ ] Correctness - A cop that alerts when incorrect coding behavior is being used
- [ ] Sharing - A cop that alerts to missing functionality necessary for sharing cookboooks with others
- [ ] Style - A cop that alerts to a style best practice
- [ ] Simplification - A cop that alerts to ways code can be simplified through refactoring
- [ ] Effortless - A cop that alerts on code that must be resolved to move to the Effortless pattern
- [ ] Policyfiles - A cop that alerts on code that must be resolved to move to the Policyfiles pattern

### Describe the new cop:
This warning is shown when you use the old-style notification syntax. You should prefer the new-style notification syntax which has the advantage that you can notify resources that are defined later.

### What it would trigger on:

```ruby
template '/etc/www/configures-apache.conf' do
  notifies :restart, resources(:service => 'apache')
end
```

### What it would autocorrect to if applicable

```ruby
template '/etc/www/configures-apache.conf' do
  notifies :restart, 'service[apache]'
end
``

**Comment by tas50** [comment]
We fixed this already
